src/core/server/stacks/approveComment.ts

```diff
@@ -71,37 +69,4 @@ const approveComment = async (
   return result.after;
 };
 
-const enableRepliesToChildren = async (
-  commentID: string,
-  mongo: MongoContext
-) => {
-  const children = await mongo
-    .comments()
-    .find({
-      parentID: commentID,
-    })
-    .toArray();
-
-  if (children.length === 0) {
-    return;
-  }
-
-  const nonRejectedChildIDs = children
-    .filter(({ status }) => status !== GQLCOMMENT_STATUS.REJECTED)
-    .map(({ id }) => id);
-
-  const allChildIDs = children.map(({ id }) => id);
-
-  await mongo
-    .comments()
-    .updateMany(
-      { id: { $in: allChildIDs } },
-      { $set: { rejectedAncestor: false } }
-    );
-
-  await Promise.all(
-    nonRejectedChildIDs.map((id) => enableRepliesToChildren(id, mongo))
-  );
```

Comment:
The recursive nature of doing many sequential Mongo calls is also why I opted for the read-only check for rejected ancestors.

Mongo isn't atomic, so we want to try and do all the updates in one `updateMany` call if possible. It avoids race conditions.

Also the way this is laid out, for a long and deep thread, we're going to see many recursive calls to write this `rejectedAncestor` flag out to all the children.

Which is what we see in Mongo when we look at the Profiler chart:

<img width="1702" alt="Screen Shot 2022-10-04 at 9 23 28 AM" src="https://user-images.githubusercontent.com/5751504/193898588-2acbdf91-0ec0-42ef-983b-1d06da04a3fb.png">

All of that orange is the read/write updates to find by parentID and then update the flag as a moderators goes through and approve/reject comments. Unfortunately, writing a simple flag out adds a lot of traffic really quickly!